File: examples/mwalib-print-tile-info.py
#!/usr/bin/env python
#
# pymwalib examples/print-tile-info - print tile info given a metafits file.
#
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.
#
#
import argparse
import mwalib

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument(
        "-m", "--metafits", required=True, help="Path to the metafits file."
    )
    args = parser.parse_args()

    metafits_filename = args.metafits

    with mwalib.MetafitsContext(metafits_filename) as context:
        print(
            "input\tant\tpol\tname\trec\tslot\telec len"
            " (m)\tflagged?\tflavour\twhitening_filter"
        )
        for i, r in enumerate(context.rf_inputs):
            print(
                f"{i}\t{r.ant}\t{r.pol}\t{r.tile_name}\t{r.rec_number}\t{r.rec_slot_number}\t"
                f"{r.electrical_length_m}\t{1 if r.flagged else 0}\t"
                f"{r.flavour}\t{r.has_whitening_filter}"
            )

File: examples/mwalib-print-tile-info.rs
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

// Given an observation's data, verify that `mwalib` is functioning correctly
// by printing an observation context.

// run this example with:
// $ cargo run --example mwalib-print-obs-context -- --metafits metafits_filename gpuboxfilename1 gpuboxfilename2...
//
// Turn on logging with: (then rerun)
// $ export RUST_LOG=mwalib=debug
//
use anyhow::*;
use clap::Parser;

use mwalib::*;

#[derive(Parser, Debug)]
#[clap(name = "mwalib-print-obs-context", author)]
struct Opt {
    /// The path to an observation's metafits file.
    #[clap(short, long)]
    metafits: std::path::PathBuf,
    // /// Paths to the observation's gpubox files.
    // #[clap(name = "GPUBOX FILE", parse(from_os_str))]
    // files: Vec<std::path::PathBuf>,
}

fn main() -> Result<(), anyhow::Error> {
    env_logger::try_init().unwrap_or(());
    let opts = Opt::parse();
    let context = MetafitsContext::new(opts.metafits, None)?;

    println!("idx\tant\tid  \tname   \tpol\trx\tslot\tlength\trx_type\tflavour   \twhitening");
    for (i, input) in context.rf_inputs.iter().enumerate() {
        println!(
            "{i:3}\t{:3}\t{:4}\t{:7}\t{:3}\t{:2}\t{:4}\t{:+7.1}\t{:7}\t{:10}\t{:9}\t",
            input.ant,
            input.tile_id,
            input.tile_name,
            input.pol.to_string(),
            input.rec_number,
            input.rec_slot_number,
            input.electrical_length_m,
            input.rec_type.to_string(),
            input.flavour.to_string(),
            input.has_whitening_filter
        );
    }

    Ok(())
}
